internal/server/data/migrations.go

```diff
@@ -1329,3 +1330,48 @@ func addAccessKeyIssuedForKind() *migrator.Migration {
 		},
 	}
 }
+
+func storeProviderUserGroupsArray() *migrator.Migration {
+	return &migrator.Migration{
+		ID: "2023-01-26T13:37",
+		Migrate: func(tx migrator.DB) error {
+			var groupType string
+			if err := tx.QueryRow(`SELECT pg_typeof(groups) FROM provider_users LIMIT 1;`).Scan(&groupType); err != nil {
+				return err
+			}
+			if groupType == "jsonb" {
+				// this migration has already been performed
+				return nil
+			}
+			// copy the groups string to the new groupsArray column, this could be a large operation if there are a lot of users
+			type userGroups struct {
+				IdentityID uid.ID
+				ProviderID uid.ID
+				Groups     models.CommaSeparatedStrings
+			}
+			rows, err := tx.Query(`SELECT identity_id, provider_id, groups FROM provider_users`)
+			if err != nil {
+				return err
+			}
+			users, err := scanRows(rows, func(u *userGroups) []any {
+				return []any{&u.IdentityID, &u.ProviderID, &u.Groups}
+			})
+			if err != nil {
+				return err
+			}
+			if _, err := tx.Exec(`ALTER TABLE provider_users DROP COLUMN IF EXISTS groups;`); err != nil {
+				return err
+			}
+			if _, err := tx.Exec(`ALTER TABLE provider_users ADD COLUMN IF NOT EXISTS groups jsonb;`); err != nil {
```

Comment:
You should be able to use a single `ALTER TABLE`

Reply:
Just tried this and it turns out you can't drop and add in a single statement.